dart/lib/src/sentry_client.dart

```diff
@@ -131,18 +132,42 @@ class SentryClient {
       return event;
     }
 
-    if (event.exceptions?.isNotEmpty ?? false) return event;
+    if (event.exceptions?.isNotEmpty ?? false) {
+      return event;
+    }
+
+    final isolateName = getIsolateName();
+    // isolates have no id, so we use the hashCode as id
+    final isolateId = isolateName?.hashCode;
 
     if (event.throwableMechanism != null) {
-      final sentryException = _exceptionFactory.getSentryException(
+      var sentryException = _exceptionFactory.getSentryException(
         event.throwableMechanism,
         stackTrace: stackTrace,
       );
 
-      return event.copyWith(exceptions: [
-        ...(event.exceptions ?? []),
-        sentryException,
-      ]);
+      if (_options.platformChecker.isWeb) {
+        return event.copyWith(
+          exceptions: [
+            ...?event.exceptions,
+            sentryException,
+          ],
+        );
+      }
+
+      sentryException = sentryException.copyWith(threadId: isolateId);
+
+      final thread = SentryThread(
+        id: isolateId,
+        name: isolateName,
+        crashed: false,
```

Comment:
I'm setting this to `false` since an exception doesn't crash the app, but not sure if that's a good default.

Reply:
The docs is off here https://develop.sentry.dev/sdk/event-payloads/threads/#attributes
That means this is the thread that actually caused the error/exception, it does not need to be a hard crash.

If the App crashed or not, look at https://develop.sentry.dev/sdk/event-payloads/exception/#attributes-1 `handled`

Reply:
Okay, I'll change it

Reply:
In that case, the Dart isolate and the Android thread probably shouldn't be connected to the same thread as suggested in https://github.com/getsentry/sentry-dart/pull/847#discussion_r857608217